engine/apps/slack/scenarios/resolution_note.py

```diff
@@ -372,18 +373,26 @@ def get_resolution_note_blocks(self, resolution_note):
         return blocks
 
 
-class ResolutionNoteModalStep(CheckAlertIsUnarchivedMixin, scenario_step.ScenarioStep):
+class ResolutionNoteModalStep(CheckAlertIsUnarchivedMixin, AlertGroupActionsMixin, scenario_step.ScenarioStep):
+    REQUIRED_PERMISSIONS = [RBACPermission.Permissions.CHATOPS_WRITE]
     RESOLUTION_NOTE_TEXT_BLOCK_ID = "resolution_note_text"
     RESOLUTION_NOTE_MESSAGES_MAX_COUNT = 25
 
     def process_scenario(self, slack_user_identity, slack_team_identity, payload, data=None):
-        AlertGroup = apps.get_model("alerts", "AlertGroup")
+        if data:
```

Comment:
worth to add the comment that this is for backwards compatibility

Reply:
addressed in https://github.com/grafana/oncall/pull/2023/commits/9bf93ed6e23c376148e2009afddf4c833a77a239